src/gluonnlp/data/sampler.py

```diff
@@ -302,6 +308,8 @@ def __init__(self, lengths: Union[Sequence[int], Sequence[Sequence[int]]],
         self._batches = []
         self._shuffle = shuffle
         self._rng = np.random.RandomState(seed)
+        self._num_parts = num_parts
+        self._part_index = part_index
```

Comment:
You may use `SplitSampler` + `BoundedBudgetSampler`. There is the 'num_parts' and 'part_index' flags in `SplitSampler`

Reply:
After offline discussion, it makes more sense to put num_parts and part_index inside.